fluture.js

```diff
@@ -617,6 +658,19 @@
     return f;
   }
 
+  //Creates a dispatcher for a binary method, but takes the object first rather than last.
+  function createInvertedBinaryDispatcher(method){
+    const f = function invertedBinaryDispatch(m, a, b){
+      if(arguments.length === 1) return unaryPartial(f, m);
+      if(arguments.length === 2) return binaryPartial(f, m, a);
+      if(m && typeof m[method] === 'function') return m[method](a, b);
```

Comment:
Thanks for the explanation. As you know I'm in favour of descriptive error messages. :)